src/testsets.jl

```diff
@@ -2,23 +2,70 @@ import Test: Test, finish, record
 
 using Test
 using Test: AbstractTestSet
+using ReTestItems: JUnitTestSuites, JUnitTestSuite, JUnitTestCase, write_junit_file, junit_record!
+import ReTestItems
 
-# Wraps a DefaultTestSet and adds a list of Tasks for concurrent tests
-mutable struct ConcurrentTestSet <: Test.AbstractTestSet
+mutable struct RAITestSet <: Test.AbstractTestSet
     dts::Test.DefaultTestSet
-    tests::Vector{Task}
+    report::Bool
+    distributed::Bool
+    distributed_tests::Vector{Task}
+    junit::Union{JUnitTestSuites, JUnitTestSuite}
```

Comment:
Does the junit output need to be part of this, or could it be a separate testset wrapper?
```
@testset JUnitTestSet begin
    @testset RAITestSet begin
...
```
At the cost of more indentation, that would separate responsibility and keep them simpler. If we're generating more than what is contained in a standard testset then this is much less practical.

Reply:
The `RAITestSet` is used at all of the non-test_rel levels and is needed for rolling up reports. By default test sets inherit from their parent, so this takes advantage of that, so we only need to mark the outer test set as `RAITestSet` and all the inner ones can be un-annotated.

We could have it just at the root, but then we'd need to maintain another testset type, which is unappealing to me and we'd need to often look to the root of the tree to figure out what's happening and whether to generate junit (and how to record it). The fact that `TestRelTestSet` are nested in `RAITestSet` makes all of this a lot easier.

I'm not sure I'm following the practicality concern - could you expand on that?

Reply:
The practicality concerns of keeping it separate that I was thinking of are just what you're describing. RAITestSet doing it all seems the better choice.